### PR TITLE
Fixes type errors and introduces numpy typing

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

File: varats/varats/plots/bug_fixing_relations.py [hunk 1]
```diff
@@ -207,6 +209,7 @@ def _generate_line_data(
                 bug_introduction] = NodeType.INTRODUCING_FIX if commit_type[
                     bug_introduction] == NodeType.FIX else NodeType.INTRODUCTION
 
+            # HERE
```
Comment (boehmseb) [inline varats/varats/plots/bug_fixing_relations.py]: What does this mean?

**Review by boehmseb — CHANGES_REQUESTED** [review]
There is one comment which I don't understand. The rest looks fine.

File: varats/varats/plots/bug_fixing_relations.py (hunk 1 above)
Comment (vulder) [reply varats/varats/plots/bug_fixing_relations.py]: nothing :rofl:

Comment (vulder) [inline varats/varats/plots/bug_fixing_relations.py]:
```suggestion
```

**Review by boehmseb — APPROVED** [review]

**Pull request merged**